tests/test_csc.py

```diff
@@ -395,17 +403,350 @@ async def doit():
+                # positions will get published
+                if atspectrograph:
+                    filter_name, filter_name2 = 'test_filt1', 'empty_1'
+                    filter_position, filter_position2 = 1, 2
+                    filter_focus_offset, filter_focus_offset2 = 0.03, 0.0
+                    filter_central_wavelength, filter_central_wavelength2 = 707.0, 700
+
+                    disperser_name, disperser_name2 = 'test_disp1', 'empty_2'
+                    disperser_position, disperser_position2 = 1, 2
+                    disperser_focus_offset, disperser_focus_offset2 = 0.1, 0.0
+
+                else:
+                    filter_name, filter_name2 = '', ''
+                    filter_focus_offset, filter_focus_offset2 = 0.0, 0.0
+                    filter_central_wavelength, filter_central_wavelength2 = 0.0, 0.0
+
+                    disperser_name, disperser_name2 = '', ''
+                    disperser_focus_offset, disperser_focus_offset2 = 0.0, 0.0
+
+                if atspectrograph and online_before_ataos:
+                    logger.debug('Loading filter and dispersers before enabling ATAOS')
+                    # Bring spectrograph online and load filter/disperser
+                    harness.atspectrograph.evt_summaryState.set_put(summaryState=salobj.State.ENABLED)
+
+                    harness.atspectrograph.evt_reportedFilterPosition.set_put(
+                        name=filter_name,
+                        position=filter_position,
+                        centralWavelength=filter_central_wavelength,
+                        focusOffset=filter_focus_offset
+                    )
+                    harness.atspectrograph.evt_reportedDisperserPosition.set_put(
+                        name=disperser_name,
+                        position=disperser_position,
+                        focusOffset=disperser_focus_offset
+                    )
+                    await asyncio.sleep(1)
+                # If the atpneumatics and atspectrograph sending events then this command times out
+                # when using the default timeout, therefore it is extended here to account for that
+                # case.
+
+                await salobj.set_summary_state(harness.aos_remote, salobj.State.ENABLED, timeout=60)
+                self.assertEqual(harness.csc.summary_state, salobj.State.ENABLED)
+
+                # send elevation/azimuth positions
+                azimuth = np.random.uniform(0., 360.)
+                # make sure it is never zero because np.random.uniform is [min, max)
+                elevation = 90. - np.random.uniform(0., 90.)
+
+                await publish_mountEncoders(harness, azimuth, elevation, ntimes=5)
+
+                # Start the ATAOS correction loop
+                # this would fail if the spectrograph isn't online!
+                if correction_loop is True and online_before_ataos is True:
```

Comment:
Why are you using `correction_loop is True ` etc here? These are booleans right? and you used them before as such. I think it would be better to just do

```
if correction_loop and online_before_ataos:
```